proto/agent.go

```diff
@@ -107,44 +109,41 @@ func (a *Agent) loop(ctx context.Context) error {
 	}
 }
 
-func (a *Agent) runTask(ctx context.Context, tc *taskContext) error {
-	ctx, cancel := context.WithCancel(ctx)
-	tc.logger = a.comm.GetLoggerProducer(ctx, tc.task)
-	metrics := &metricsCollector{
-		comm:     a.comm,
-		taskData: tc.task,
-	}
-
-	if err := metrics.start(ctx); err != nil {
-		return errors.Wrap(err, "problem setting up metrics collection")
-	}
+func (a *Agent) resetLogging(ctx context.Context, tc *taskContext) error {
+	tc.logger = a.comm.GetLoggerProducer(context.TODO(), tc.task)
```

Comment:
What's the reason for making this uncancelable compared to what it was before?

Reply:
Closing the logger cancels the context, and otherwise there's no need to cancel it, I don't think. I tried a couple different ways of implementing the cancel, but they either panic (because of a race with Close) or the race detector alerts on them. I think this is okay for now.